drivers/docker/coordinator.go

```diff
@@ -0,0 +1,416 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"sync"
+	"time"
+
+	docker "github.com/fsouza/go-dockerclient"
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+var (
+	// createCoordinator allows us to only create a single coordinator
+	createCoordinator sync.Once
+
+	// globalCoordinator is the shared coordinator and should only be retrieved
+	// using the GetDockerCoordinator() method.
+	globalCoordinator *dockerCoordinator
```

Comment:
Since only 1 driver instance is created per plugin these don't have to be global anymore!